### DISCUSSION TITLE
How does the FrameAnimation3d class work

### DISCUSSION THREAD

**Discussion by BellVirtual** [pokepetter/ursina#168] [Q&A]
I have been doing some developing with the ursina engine but I wanted to ad some animations to my 3d models. However I noticed after scouring the internet that no one has written any code using the FrameAnimation3d class and there are no samples files using it. Does any one out there no to to import 3d animations into ursina.

**Comment by pokepetter** [comment]
Loads an obj sequence as a frame animation.
So if you have some frames named run_cycle_000.obj, run_cycle_001.obj, run_cycle_002.obj and so on,
you can load it like this: FrameAnimation3d('run_cycle_')

FrameAnimation3d('blob_animation_')

**Reply by BellVirtual** [reply]
Wait so if I want to load a 3D animation I have to have every frame of the animation in a separate  .obj file?
Is there any way to load a 3D animation from a single .blend or .obj file?

**Reply by pokepetter** [reply]
.obj doesn't support animation.
You can use panda3d's Actor class and export your file to one of the supported formats

**Reply by BellVirtual** [reply]
After many trials and tribulations I implemented the panda 3d actor class.
However I can only load .egg files using that class which is really clumsy to deal with.
Would you maybe consider adding a class to ursina that loads 3d animations from .blend files.
I really really like your engine and am happy to hear from you, I honestly think this is the only thing holding ursina back.

**Reply by korakoe** [reply]
I would also like to see this support gltf animations that are embedded
